kiln/devnets/setup.sh

```diff
@@ -127,7 +151,13 @@ then
 	echo "you can watch observe the client logs at the respective terminals"
 	echo "use ctl + c on any of these three (including this) terminals to stop the process"
 	echo "waiting ..."
-	wait -n $elPid $clPid
+	if [ $platform == 'Darwin' ]
+	then # macOs ships with an old version of bash with wait that does not have the -n flag
+	  wait $elPid
+	  wait $clPid
```

Comment:
Is there a way to wait for both the processes together in Mac ? Because the intend here is to check if any process exits then do cleanup

Reply:
> Is there a way to wait for both the processes together in Mac ?

This is exactly what happens. The non-mac version which uses `-n` flag, modifies the `wait` command to only wait for a single job from the provided process ID or job specs to finish before returning its exit status [source](https://linuxhint.com/bash-wait-command-linux/#:~:text=Using%20%E2%80%93n%20option%3A,if%20no%20arguments%20are%20provided). 

I tested this out also and confirmed this behaviour of `-n` flag on a linux server.

So the mac version won't short-circuit, while the -n flag will. There would probably be a _hackish_ way to have the same circuiting behaviour for mac, but I consider losing that behaviour to be acceptable as it does not radically change the utility of the script.